### PR TITLE
++ #2530 Fixing Neo integ tests & adding more input validations 

### PR DESCRIPTION
**Pull request by HappyAmazonian**
## Description ##
did a format based on https://github.com/deepjavalibrary/djl-serving/pull/2530

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/docker/partition/sm_neo_dispatcher.py [hunk 1]
```diff
@@ -53,10 +53,26 @@ def _get_serving_properties(self) -> dict:
...
-                and self.properties.get("option.tensor_parallel_degree")
-                is not None)
+        if self.properties.get("option.load_format") in VALID_LOAD_FORMATS:
+            if self.properties.get("option.quantize"):
+                raise ValueError(
+                    f"Sharding & Quantization are mutually exclusive. "
+                    f"Please Quantize first & then Shard. "
+                    f"Received load_format={self.properties.get('option.load_format')} "
+                    f"& quantize={self.properties.get('option.quantize')}.")
+            if not int(self.properties.get("option.tensor_parallel_degree")):
```
Comment (sindhuvahinis) [inline serving/docker/partition/sm_neo_dispatcher.py:63]: int(None) -> would throw error.

File: engines/python/setup/djl_python/properties_manager/lmi_dist_rb_properties.py [hunk 2]
```diff
@@ -82,6 +86,14 @@ def validate_speculative_and_lora(self):
             )
         return self
 
+    @model_validator(mode='after')
+    def validate_speculative_and_fml(self):
+        if self.load_format == LmiDistLoadFormats.sagemaker_fast_model_loader and self.speculative_draft_model:
```
Comment (sindhuvahinis) [inline engines/python/setup/djl_python/properties_manager/lmi_dist_rb_properties.py:91]: self.load_format seems to be str. So you might need to change to `self.load_format == LmiDistLoadFormats.sagemaker_fast_model_loader.value`.  Or change the load_format type from str to LmiDistLoadFormats enum. I prefer the latter.

Comment (HappyAmazonian) [reply engines/python/setup/djl_python/properties_manager/lmi_dist_rb_properties.py]: it seems self.load_format  is used in other places. Let me do the .value change first.

File: engines/python/setup/djl_python/properties_manager/lmi_dist_rb_properties.py [hunk 3]
```diff
@@ -82,6 +86,14 @@ def validate_speculative_and_lora(self):
             )
         return self
 
+    @model_validator(mode='after')
+    def validate_speculative_and_fml(self):
+        if self.load_format == LmiDistLoadFormats.sagemaker_fast_model_loader.value and self.speculative_draft_model:
+            raise AssertionError(
+                f"Cannot enable sagemaker_fast_model_loader and speculative decoding at the same time"
```
Comment (siddvenk) [inline engines/python/setup/djl_python/properties_manager/lmi_dist_rb_properties.py:93]: What is the customer facing name for fast_model_loader? We should probably use the recognizable name here so customers know exactly what the incompatibility is. 

Once we have that, I would suggest changing this to `speculative decoding and <fast model loading name> are currently not compatible. Disable speculative decoding, or use standard checkpoints for the target model`

**Review by siddvenk — APPROVED** [review]

**Pull request merged**